PhysicsTools/TensorFlow/src/NTSession.cc

```diff
@@ -523,13 +498,10 @@ Status NTSession::Run(const RunOptions& run_options,
     run_state.status.Update(errors::Cancelled("Run call was cancelled"));
```

Comment:
around L510 the upstream has the following change

```diff
     return errors::Cancelled("Run call was cancelled");
   }

+  Executor::Args::Runner default_runner = [this,
+                                           pool](Executor::Args::Closure c) {
+    SchedClosure(pool, std::move(c));
+  };
   for (const auto& item : executors_and_keys->items) {
+    // TODO(zhengxq): support partial run.
+    // TODO(zhengxq): if the device picks its own threadpool, we need to assign
+    //     less threads to the main compute pool by default.
+    thread::ThreadPool* device_thread_pool =
+        item.device->tensorflow_device_thread_pool();
+    if (!device_thread_pool) {
+      args.runner = default_runner;
+    } else {
+      args.runner = [this, device_thread_pool](Executor::Args::Closure c) {
+        SchedClosure(device_thread_pool, std::move(c));
+      };
+    }
     item.executor->RunAsync(args, barrier->Get());
   }

```

should it be made here?

Reply:
See previous comment. The if-clause in the loop body should always be true, which is why I moved the `args.runner = default_runner;` line outside the loop and also removed the `default_runner` variable.

Reply:
we should remove any optimizations with respect to the upstream code except for the functional changes that we need. This will be very useful when the upstream is refactored or updated in the future.

If the improvement is valid for the upstream, perhaps it's better to submit a PR there.

Reply:
IIUC, there is still some optimization present here from our side, which leads to the change in the loop body.
Please either restore possibly redundant change or add a clear comment about the modification made from CMSSW side.

Reply:
Ok, makes sense. I will update that part.